app/controllers/income/UpdateIncomeNextYearController.scala

```diff
@@ -81,16 +76,24 @@ trait UpdateIncomeNextYearController extends TaiBaseController
             }
           } else {
             Future.successful(NotFound(error4xxPageWithLink(Messages("global.error.pageNotFound404.title"))))
-          }.recoverWith{
-            case e: Exception => {
-              Logger.warn(e.getMessage)
-              Future.successful(InternalServerError(error5xx(Messages("tai.technical.error.message"))))
-            }
           }
   }
 
   def confirm (employmentId: Int): Action[AnyContent] = ???
-  def success (employmentId: Int): Action[AnyContent] = ???
+  def success (employmentId: Int): Action[AnyContent] = authorisedForTai(personService).async {
+    implicit user =>
+      implicit person =>
+        implicit request =>
+          if(cyPlusOneEnabled){
+            ServiceCheckLite.personDetailsCheck {
+              updateNextYearsIncomeService.get(employmentId, Nino(user.getNino)) map { model =>
```

Comment:
We need to have a flush of the cache at this point. It could be a service method that gets, resets and returns the model?

Reply:
We reset at the start of the journey already. Do we need to reset again at the end?

Reply:
Done